configurations/dockerYamlTemplate.go

```diff
@@ -91,6 +93,16 @@ func convertMapToMap(mapInterface map[string]interface{}) map[string]string {
 	return mapString
 }
 
+// Parses the programmingLanguage instruction from yaml and returns an instance of `programmingLanguage`.
+func parseProgrammingLanguage(data map[string]interface{}) instruction {
+	convertedData := convertMapToMap(data)
+	var langObj programmingLanguage
+	langObj.Name = convertedData["name"]
```

Comment:
what is the guarantee that `convertedData` is going to have "name" as a key?

Reply:
Added check

Reply:
This again translates to the parsing of the yaml file.
For now, the check is fine but adding checks like this results in unmaintainable code.
This issue should also get resolved in the PR for Issue #17.

Reply:
Yes